### PR TITLE
Extend use_state docstring

### PR DESCRIPTION
**Pull request by liamhuber**
@larsoner [is right that the docstring should be updated](https://github.com/h5io/h5io/pull/77#issuecomment-1930700708). Since I'm probably the keenest one for a new version release, I thought I'd go ahead and draft the changes. I'm not married to this particular wording or anything, so feel free to commit changes! @pmrv @jan-janssen

### CODE REVIEW

File: h5io/_h5io.py [hunk 1]
```diff
@@ -107,9 +107,18 @@ def write_hdf5(
         To accelerate the read and write performance of small dictionaries and
         lists they can be combined to JSON objects and stored as strings.
     use_state: bool
-        To store objects of unsupported types the __getstate__() method is used
-        to retrieve a dictionary which defines the state of the object and store
-        the content of this dictionary in the HDF5 file. (requires python >=3.11)
+        To store objects of unsupported types the `__getstate__()` and other dunder
+        methods (`__reduce__()`, `__getnewargs__()`  etc.) are used to  retrieve a
+        dictionary which defines the state of the object and store the content of this
+        dictionary in the HDF5 file. The interface strives to behave in like `pickle`,
```
Comment (jan-janssen) [inline h5io/_h5io.py:113]:
```suggestion
        dictionary in the HDF5 file. The interface strives to behave like `pickle`,
```

**Review by jan-janssen — APPROVED** [review]
Looks good to me

**Comment by larsoner** [comment]
Thanks @liamhuber

**Comment by liamhuber** [comment]
@larsoner would it be possible to get a new release now that this content is in?

**Comment by larsoner** [comment]
Cut a release at https://github.com/h5io/h5io/releases/tag/h5io-0.2.2, should land on PyPI and conda-forge soon but let me know if they don't!

**Comment by liamhuber** [comment]
Awesome, thanks @larsoner!!

**Pull request merged**